napari/layers/base/base.py

```diff
@@ -568,12 +568,34 @@ def affine(self, affine):
 
     @property
     def translate_grid(self):
-        """list: Factors to shift the layer by."""
-        return self._transforms['world2grid'].translate
+        warnings.warn(
+            trans._(
+                "translate_grid will become private in v0.4.14. See Layer.translate or Layer.data_to_world() instead.",
```

Comment:
I try to support at least 3 last napari releases, Short deprecation time is not helping here.